theatre/studio/src/StudioStore/StudioStore.ts

```diff
@@ -141,46 +142,62 @@ export default class StudioStore {
                   .getValue(),
               )
 
-              forEachDeep(
-                value,
-                (v, pathToProp) => {
-                  if (typeof v === 'undefined' || v === null) {
-                    return
-                  }
-
-                  const propAddress = {...root.address, pathToProp}
-
-                  const trackId = get(
-                    sequenceTracksTree,
-                    pathToProp,
-                  ) as $FixMe as SequenceTrackId | undefined
-
-                  if (typeof trackId === 'string') {
-                    const propConfig = get(
-                      root.template.config.props,
-                      pathToProp,
-                    ) as PropTypeConfig | undefined
-                    if (propConfig?.sanitize) v = propConfig.sanitize(v)
-
-                    const seq = root.sheet.getSequence()
-                    seq.position = seq.closestGridPosition(seq.position)
-                    stateEditors.coreByProject.historic.sheetsById.sequence.setKeyframeAtPosition(
-                      {
-                        ...propAddress,
-                        trackId,
-                        position: seq.position,
-                        value: v as $FixMe,
-                        snappingFunction: seq.closestGridPosition,
-                      },
-                    )
-                  } else {
-                    stateEditors.coreByProject.historic.sheetsById.staticOverrides.byObject.setValueOfPrimitiveProp(
-                      {...propAddress, value: v},
-                    )
-                  }
-                },
-                getPointerParts(pointer as Pointer<$IntentionalAny>).path,
-              )
+              const propConfig = get(
+                root.template.config.props,
+                path,
+              ) as PropTypeConfig
+
+              const setStaticOrKeyframeProp = <T>(
+                value: T,
+                path: PathToProp,
+              ) => {
+                if (typeof value === 'undefined' || value === null) {
+                  return
+                }
+
+                const propAddress = {...root.address, pathToProp: path}
+
+                const trackId = get(sequenceTracksTree, path) as $FixMe as
+                  | SequenceTrackId
+                  | undefined
+                if (typeof trackId === 'string') {
+                  const propConfig = get(root.template.config.props, path) as
+                    | PropTypeConfig
+                    | undefined
+                  if (propConfig?.sanitize) value = propConfig.sanitize(value)
```

Comment:
Now that we're decorating within sanitize, the value is not neccesarily serializable. This can remain for now, but maybe let's leave a `// TODO` that we need to `validate()` here instead of `sanitize()`.

Reply:
That decorating would have to do something really silly, like introduce circular references for that to happen 🤔 Or what kind of issue were you thinking exactly?

Reply:
Let's say `sanitize()` (or decorate) returns a class instance. That's not straighforward to serialize.

Reply:
yep, well, we are going to overhaul the validate/sanitize/decode/decorate stuff anyway, so I'd leave it like this for now, will add a comment